src/mongocxx/result/insert_many.hpp

```diff
@@ -59,11 +57,15 @@ class MONGOCXX_API insert_many {
     ///
     /// @return The values of the _id field for inserted documents.
     ///
-    std::map<std::size_t, bsoncxx::document::element> inserted_ids();
+    id_map inserted_ids();
 
    private:
+    friend collection;
+
+    insert_many(result::bulk_write result, id_map inserted_ids);
```

Comment:
I believe this is addressed in your other review, but this method when made C++ private should also be made ABI private with a MONGOCXX_PRIVATE annotation